spark/src/main/scala/io/ddf/spark/content/RDDRow2REXP.scala

```diff
@@ -44,7 +44,17 @@ class RDDROW2REXP(@transient ddf: DDF) extends ConvertFunction(ddf) {
                       buffer += row.getInt(i)
                     }
                   }
-                  case ColumnType.DOUBLE => {
+
+                  case ColumnType.BIGINT => {
+                    val buffer = arrayBufferColumns(i).asInstanceOf[ArrayBuffer[Double]]
+                    if (row.isNullAt(i)) {
+                      buffer += REXPDouble.NA
+                    } else {
+                      buffer += row.getLong(i).toDouble
+                    }
+                  }
+
+                  case ColumnType.DOUBLE | ColumnType.BIGINT => {
```

Comment:
Yes, R has no long/bigint data type. So we have to convert from/to double
for DDF's BigInt type.

On Thu, Dec 17, 2015 at 8:37 AM, Hai-Anh Trinh <notifications@github.com>
wrote:

> In spark/src/main/scala/io/ddf/spark/content/RDDRow2REXP.scala
> <https://github.com/ddf-project/DDF/pull/244#discussion_r47860082>:
>
> > @@ -44,7 +44,17 @@ class RDDROW2REXP(@transient ddf: DDF) extends ConvertFunction(ddf) {
> >                        buffer += row.getInt(i)
> >                      }
> >                    }
> > -                  case ColumnType.DOUBLE => {
> > +
> > +                  case ColumnType.BIGINT => {
> > +                    val buffer = arrayBufferColumns(i).asInstanceOf[ArrayBuffer[Double]]
> > +                    if (row.isNullAt(i)) {
> > +                      buffer += REXPDouble.NA
> > +                    } else {
> > +                      buffer += row.getLong(i).toDouble
> > +                    }
> > +                  }
> > +
> > +                  case ColumnType.DOUBLE | ColumnType.BIGINT => {
>
> oh is this because R does not have such type?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ddf-project/DDF/pull/244/files#r47860082>.
>



-- 
If life gives you lemons, make lemonade.